src/App.tsx

```diff
@@ -19,55 +63,59 @@ export const App: React.FC = () => {
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              value={query}
+              onChange={handleQueryChange}
+              onClick={handleInputClick}
+              onFocus={handleInputFocus}
             />
           </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
+          {isDropdownOpen && (
+            <div
+              className="dropdown-menu"
+              role="menu"
+              data-cy="suggestions-list"
+            >
+              <div className="dropdown-content">
+                {filteredPeople.map(person => (
+                  <div
+                    className={classNames('dropdown-item', {
+                      active: selectedPerson === person,
+                    })}
+                    data-cy="suggestion-item"
+                    key={peopleFromServer.indexOf(person)}
+                    onClick={() => onSelected(person)}
+                  >
+                    <p
+                      className={classNames({
+                        'has-text-link': person.sex === 'm',
+                        'has-text-danger': person.sex === 'f',
```

Comment:
Create a enum for 'm' and 'f' and use it here

Reply:
Create a enum for 'm' and 'f' and use it here